modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -553,6 +554,144 @@
       (log/debugf e "Query:\n%s" native-form)
       native-form)))
 
+(defmethod driver/create-table! :mongo
+  [_driver database-id table-name _column-definitions & {:keys [primary-key]}]
+  ;; MongoDB collections are created implicitly when first document is inserted
+  ;; We can create an empty collection explicitly if needed
+  (mongo.connection/with-mongo-database [^MongoDatabase db database-id]
+    (.createCollection db (name table-name))
+    ;; Create indexes for any primary key fields
+    (when primary-key
+      (doseq [pk-field primary-key]
+        (mongo.util/create-index
+         (mongo.util/collection db (name table-name))
+         {pk-field 1})))))
+
+(defmethod driver/drop-table! :mongo
+  [_driver db-id table-name]
+  (mongo.connection/with-mongo-database [^MongoDatabase db db-id]
+    (some-> (mongo.util/collection db (name table-name))
+            .drop)))
+
+(defmethod driver/rename-tables!* :mongo
+  [_driver db-id sorted-rename-map]
+  (mongo.connection/with-mongo-database [^MongoDatabase db db-id]
+    (doseq [[old-table-name new-table-name] sorted-rename-map]
```

Comment:
Add a TODO to check whether we can make this atomic with locks, link that issue (sorry gonna keep spamming)